app/controllers/provider_interface/courses_controller.rb

```diff
@@ -0,0 +1,36 @@
+module ProviderInterface
+  class CoursesController < ProviderInterfaceController
+    before_action :set_application_choice
+    before_action :redirect_to_application_page_unless_feature_flag_is_active
+
+    def edit
+      redirect_to provider_interface_application_choice_course_path(@application_choice)
+    end
+
+  private
+
+    def change_course_store
+      key = "change_course_wizard_store_#{current_provider_user.id}_#{@application_choice.id}"
+      WizardStateStores::RedisStore.new(key: key)
+    end
+
+    def action
+      'back' if !!params[:back]
+    end
+
+    def available_providers
+      query_service.available_providers
+    end
+
+    def query_service
+      @query_service ||= GetChangeOfferOptions.new(
+        user: current_provider_user,
+        current_course: @application_choice.current_course,
+      )
+    end
+
+    def redirect_to_application_page_unless_feature_flag_is_active
+      redirect_to provider_interface_application_choice_path(@application_choice) unless FeatureFlag.active?(:change_course_details_before_offer)
```

Comment:
We'll need to add a check that they can make decisions as well, but can come in a different pr

Reply:
[This check](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/controllers/provider_interface/provider_interface_controller.rb#L1110) can be reused in a before_action callback

Reply:
Will address in a later PR!